libs/server-sdk/src/data_interfaces/system/isystem.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "../store/istore.hpp"
+
+namespace launchdarkly::server_side::data_interfaces {
+
+/**
+ * \brief ISystem obtains data used for flag evaluations and makes it available
+ * to other components.
+ */
+class ISystem : public IStore {
```

Comment:
Not sure about this name.

Reply:
The `ISystem` name? I guess it should actually be `IDataSystem` (but you might have objections to that as well? It's from the Data System spec work.)

The intention behind this interface is to reflect that it might be more than an `IStore`. Perhaps that's overengineering, but it does add the `Initialize()` method, which `IStore()` doesn't have.

Reply:
It is like IManager. If you use the pattern more than once, then you have already name squatted. I know it has a namespace, but it would end up very confusing when actually reading code.

IDataSystem doesn't have that problem, at least not really for our SDKs.

Reply:
Ah,  I originally had all of the interfaces with a `Data` prefix, like `IDataStore`, `IDataDestination`, etc. 

But when put in the `data_interface` namespace, these all seemed quite redundant. I've been using the namespace prefixes throughout in the callsites instead. `data_interfaces::IDataSystem` might be fine though since it is the top-level concept (and like you said, `System` is insanely generic.)

Reply:
Will rename to `IDataSystem` in subsequent PR.